### PR TITLE
feat: skip e2e based off build tags

### PR DESCRIPTION
**Pull request by george-e-shaw-iv**
**What this PR does / why we need it**: 
* Conditionally skips e2e tests if no `or_e2e` build tags are found.

**JIRA ID**: XX-XX

**Notes for your reviewer**:

### CODE REVIEW

File: e2e/e2e.go
Comment (george-e-shaw-iv) [inline e2e/e2e.go:225]: lol guess who forgot to close the file again

Comment (george-e-shaw-iv) [reply e2e/e2e.go:225]: wait i don't have to in this case. played myself.

Comment (jaredallard) [reply e2e/e2e.go:225]: yeah I was about to comment, I don't think reading it into memory is a great idea if we can avoid it

**Review by jaredallard — APPROVED** [review]
This is a great QoL improvement, thanks! We can address `os.ReadFile` now, or later, not a huge deal

File: e2e/e2e.go
Comment (plisy) [inline e2e/e2e.go:215]: Is this because we are still on go 1.15 and WalkDir was introduced in 1.16?

**Review by plisy — APPROVED** [review]

File: e2e/e2e.go
Comment (george-e-shaw-iv) [reply e2e/e2e.go:215]: yeah we're not completely switched over to 1.16 as a whole so i think it's safe to continue to use filepath.Walk

File: e2e/e2e.go
Comment (vk-outreach) [inline e2e/e2e.go:204]: I think it might be simpler to use [build.ImportDir](https://pkg.go.dev/go/build#Context.ImportDir) and inspect the [Package.AllTags](https://pkg.go.dev/go/build#Package).   If you decide to just keep the current structure, I suggest atleast using [constraint](https://pkg.go.dev/go/build/constraint@go1.16.6) to check if a line is a build constraint.

Comment (george-e-shaw-iv) [reply e2e/e2e.go:204]: Oh I didn't see this before hitting the merge button. I'll change the code to use the `build.ImportDir` function whenever I get some free time next, that seems like a better way to do this, I agree. I'll create a ticket to track this

Comment (george-e-shaw-iv) [reply e2e/e2e.go:204]: https://outreach-io.atlassian.net/browse/DTSS-605

**Pull request merged**
